### ISSUE TITLE
Slow performance after upgrade from goracle to godror

### ISSUE THREAD

**Issue by RenanAngeliniFerrari** [godror/godror#199]
**Describe the bug**

When starting a loop on the result of a query and inside this loop start another loop on the result of another query, as time goes by it passes, it comes to a point where an enormous slowness begins.

When running the program I mentioned below, the program starts at 
time.Date(2021, time.November, 22, 16, 2, 1, 544104100, time.Local) and
starts the slowness in 
time.Date(2021, time.November, 22, 16, 7, 5, 328730400, time.Local).

Even though I must mention Rows.Close(), Rows.Next() itself should be running since the query 'select 1 from dual' returns no error.

**To Reproduce**

```
package main

import _ "github.com/godror/godror"
import "database/sql"
import "time"

func main() {

  var DataBase *sql.DB
  var Rows *sql.Rows
  var Rows2 *sql.Rows

  DataBase, _ = sql.Open("godror","teste/teste@localhost:1521/terra")  

  for {  

    println(time.Now().GoString())

    Rows,_ = DataBase.Query("select 1 from dual")

    for Rows.Next() {

      for i := 0; i < 500; i++ {
      
        Rows2,_ = DataBase.Query("select 1 from dual")

        for Rows2.Next() {

        }
      }
    } 

    time.Sleep(1 * time.Second)
  }
}
```

**Expected behavior**
I think that the slowness should not be occurring, even if the reason be to not declare Rows.Close(), since, the Rows.Next() itself should do this work when there are no errors.

**Your oracle client version**
e.g. 11.2.0.4

**Your godror version**
e.g. 0.29.0

**Your go version**
e.g. 1.17.2

**Your gcc version**
e.g. 8.1.0

**Machine (please complete the following information):**
 - OS: Windows Pro
 - Architecture: x64
 - Version: 10

**Additional context**

**Comment by tgulacsi** [comment]
I cannot reproduce it - TestLoopInLoop_199 in z_test.go at f8b2b1c produces
```
=== RUN   TestLoopInLoop_199                                                                   
    z_test.go:4340: 462.059305ms                                                               
    z_test.go:4340: 429.240121ms                                                                   z_test.go:4340: 460.055773ms                                                               
    z_test.go:4340: 431.371632ms                                                               
    z_test.go:4340: 460.127403ms                                                               
    z_test.go:4340: 432.055959ms                                                               
    z_test.go:4340: 463.009483ms                                                               
    z_test.go:4340: 427.476793ms                                                               
    z_test.go:4340: 462.118641ms                                                               
    z_test.go:4340: 428.229692ms                                                               
    z_test.go:4340: 459.482725ms                                                               
    z_test.go:4340: 431.950263ms               
    z_test.go:4340: 459.063886ms                                                               
    z_test.go:4340: 420.39308ms                                                                
    z_test.go:4340: 463.617242ms                                                               
    z_test.go:4340: 427.850127ms                                                               
    z_test.go:4340: 462.052706ms               
    z_test.go:4340: 426.776389ms                                                               
    z_test.go:4340: 460.974206ms               
    z_test.go:4344: first: 462.059305ms last: 460.974206ms avg: 423.395271ms
--- PASS: TestLoopInLoop_199 (9.44s)                                     
```
Quite constant results.

**Comment by RenanAngeliniFerrari** [comment]
This test you ran is on version 1.14 of go, right? I will download it and run the test again. Tomorrow post the result. Thanks for the help.

**Comment by tgulacsi** [comment]
go version go1.17.2 linux/amd64
    Client: 21.3.0.0.0 Timezone: Local
    Server: 19.13.0.1.0 [Oracle Database 19c Enterprise Edition Release 19.0.0.0.0 - Production; Version 19.13.0.1.0] Timezone: UTC

**Comment by RenanAngeliniFerrari** [comment]
tgulacsi,

I took your code and put it inside an infinite loop with a 1 second pause, printing the time at the end of the loop.

In this way, I got this result of when the service started and stopped:

Start: 09:01:33
Break: 09:06:34

Start: 09:07:54
Break: 09:12:55

The amount of time is exactly the same.

I'm thinking about something related to connecting to the bank.

Thanks for the help.
